cpp/src/text/minhash.cu

```diff
@@ -133,27 +149,26 @@ std::unique_ptr<cudf::column> minhash(cudf::strings_column_view const& input,
                                           cudf::mask_state::UNALLOCATED,
                                           stream,
                                           mr);
-  auto d_hashes = hashes->mutable_view().data<cudf::hash_value_type>();
-
-  thrust::for_each_n(
-    rmm::exec_policy(stream),
-    thrust::counting_iterator(std::size_t{0}),
-    static_cast<std::size_t>(input.size()) * static_cast<std::size_t>(cudf::detail::warp_size),
-    minhash_fn{*d_strings, seeds, width, d_hashes});
-
-  if (seeds.size() == 1) {
-    hashes->set_null_mask(cudf::detail::copy_bitmask(input.parent(), stream, mr),
-                          input.null_count());
-    return hashes;
-  }
+  auto d_hashes = hashes->mutable_view().data<hash_value_type>();
+
+  constexpr int block_size = 256;
+  cudf::detail::grid_1d grid{input.size() * cudf::detail::warp_size, block_size};
+  minhash_kernel<HashFunction><<<grid.num_blocks, grid.num_threads_per_block, 0, stream.value()>>>(
+    *d_strings, seeds, width, d_hashes);
+
+  return hashes;
+}
 
+std::unique_ptr<cudf::column> build_list_result(cudf::strings_column_view const& input,
+                                                std::unique_ptr<cudf::column>&& hashes,
+                                                cudf::size_type seeds_size,
+                                                rmm::cuda_stream_view stream,
+                                                rmm::mr::device_memory_resource* mr)
+{
   // build the offsets for the output lists column
-  auto offsets = cudf::detail::sequence(
-    input.size() + 1,
-    cudf::numeric_scalar<cudf::size_type>(0),
-    cudf::numeric_scalar<cudf::size_type>(static_cast<cudf::size_type>(seeds.size())),
-    stream,
-    mr);
+  auto const zero = cudf::numeric_scalar<cudf::size_type>(0);
+  auto const size = cudf::numeric_scalar<cudf::size_type>(static_cast<cudf::size_type>(seeds_size));
```

Comment:
I'm realizing now that most of the code is not changed from what it looked like before this PR, you just converted thrust+functor into a kernel, so no need to change but just leaving another note: isn't this also an unnecessary cast?

Reply:
You are correct. This mistake was a side-effect of refactoring and copying code around. The `seed_size` had previously come from a device-span which has sizes of type `size_t` and required casting here. That is done before calling this code so the extra cast is definitely not needed.